File: app/src/main/java/com/geowatershed/app/data/Formatting.kt
package com.geowatershed.app.data

import java.util.Locale

/** Shared display formatting for a real (or missing) GPS fix. */
fun formatCoords(latitude: Double?, longitude: Double?, accuracyMeters: Float?): String {
    if (latitude == null || longitude == null) return "GPS unavailable"
    val accuracy = accuracyMeters?.let { " · ±${it.toInt()} m" }.orEmpty()
    return String.format(Locale.US, "%.5f N, %.5f E", latitude, longitude) + accuracy
}

File: geowatershed_flutter/lib/data/formatting.dart
/// Shared display formatting for a real (or missing) GPS fix.
String formatCoords(double? latitude, double? longitude, double? accuracyMeters) {
  if (latitude == null || longitude == null) return 'GPS unavailable';
  final accuracy = accuracyMeters == null ? '' : ' · ±${accuracyMeters.round()} m';
  return '${latitude.toStringAsFixed(5)} N, ${longitude.toStringAsFixed(5)} E$accuracy';
}
